db/snapshot/backup_task.hh

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "tasks/task_manager.hh"
+
+namespace s3 { class client; }
+
+namespace db {
+class snapshot_ctl;
+
+namespace snapshot {
+
+class backup_task_impl : public tasks::task_manager::task::impl {
+    snapshot_ctl& _snap_ctl;
+    shared_ptr<s3::client> _client;
+    sstring _bucket;
+    sstring _ks;
+    sstring _snapshot_name;
+    gate _uploads;
+    std::exception_ptr _ex;
+
+    void run_in_thread(sstring data_dir);
+
+protected:
+    virtual future<> run() override;
```

Comment:
nit, `virtual` here is redundant. this is a new source file. so probably we don't have do be consistent with the existing code here.